src/layers/s2-geometry-layer/s2-geometry-layer.js

```diff
@@ -0,0 +1,231 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {S2Layer} from 'deck.gl';
+import {hexToRgb} from 'utils/color-utils';
+
+import Layer from '../base-layer';
+import S2LayerIcon from './s2-layer-icon';
+import {HIGHLIGH_COLOR_3D} from 'constants/default-settings';
+
+export const S2_TOKEN_FIELDS = {
+  token: [
+    's2',
+    's2_token'
+  ]
+};
+
+export const s2RequiredColumns = ['token'];
+export const S2TokenAccessor = ({token}) => d => d[token.fieldIdx];
+
+export const S2VisConfigs = {
+  opacity: 'opacity',
+  colorRange: 'colorRange',
+  filled: {
+    type: 'boolean',
+    label: 'Fill Color',
+    defaultValue: true,
+    property: 'filled'
+  },
+
+  // height
+  enable3d: 'enable3d',
+  elevationScale: 'elevationScale',
+  elevationPercentile: 'elevationPercentile',
+  sizeRange: 'elevationRange'
+};
+
+export default class S2GeometryLayer extends Layer {
+
+  constructor(props) {
+    super(props);
+    this.registerVisConfig(S2VisConfigs);
+    this.getPositionAccessor = () => S2TokenAccessor(this.config.columns);
+  }
+
+  get type() {
+    return 's2';
+  }
+
+  get name() {
+    return 'S2'
+  }
+
+  get requiredLayerColumns() {
+    return s2RequiredColumns;
+  }
+
+  get layerIcon() {
+    return S2LayerIcon;
+  }
+
+  static findDefaultLayerProps({fields = []}) {
+    const foundColumns = this.findDefaultColumnField(S2_TOKEN_FIELDS, fields);
+    if (!foundColumns || !foundColumns.length) {
+      return {props: []};
+    }
+
+    return {
+      props: foundColumns.map(columns => ({
+        isVisible: true,
+        label: 'S2',
+        columns
+      }))
+    };
+  }
+
+  calculateDataAttribute({allData, filteredIndex}, getS2Token) {
+    const data = [];
+    for (let i = 0; i < filteredIndex.length; i++) {
+      const index = filteredIndex[i];
+      const token = getS2Token(allData[index]);
+
+      if (token) {
+        data.push({
+          // keep a reference to the original data index
+          index,
+          data: allData[index],
+          token
+        });
+      }
+    }
+    return data;
+  }
+
+  updateLayerMeta(allData, getS2Token) {
```

Comment:
we should calculate at least `bounds`, so that when people upload a s2 file, it would be centered on screen. You can ask @Pessimistress to export `getS2Polygon` from `@deck.gl/geo-layers/s2-layer/s2-utils`

Reply:
done